guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,581 @@
+[Back to Main README](../README.md)
+
+# Full GitHub Authorization with the `torii-provider`
+
+This guide walks through the process of authenticating and authorizing an Ember app with
+and for use against GitHub using `ember-simple-auth` and `torii` including a registered
+GitHub application, an unauthenticated login route, an authenticated content route, an
+AWS Lambda-based token exchange service, and use of an authorizer to inject the token
+into requests. We'll use the [ember-data-github](https://github.com/elwayman02/ember-data-github)
+addon so we don't need to create our own GitHub models, serializers, and adapters.
+
+While most of the pieces are documented in the `ember-simple-auth` and `torii`
+documentation, putting them together properly for the particular scenario has not been
+fully documented to date. Additionally, many of the sources on the general internet are
+helpful but dated or incomplete.
+
+This guide assumes you know how to start an Ember app and are familiar with the various
+files and concepts.
+
+## Registering your app with GitHub
+
+Go to your GitHub Settings, choose `OAuth Applications`, and press the `Register a new
+application` button. Fill in the fields similar to the following.
+
+![Registering an OAuth Application with GitHub](./assets/github-registering-oauth-application.png)
+
+You will likely want to register different application keys for different deployment stages
+because of the differences in the `Authorization callback URL`.
+
+Once you register your application, you will have a chance to add an application logo.
+Most importantly, you will be given a client ID and a client secret. _The client secret is
+a password and should be protected as such!_ You will use the client ID in your web
```

Comment:
Not sure whether the term _"password"_ might cause confusion here. Maybe we can say something like _"The client secret must be kept secret and must never be included e.g. in the Ember application's source"_ or so?